src/SqlVirtualMachine/SqlVirtualMachine/Common/AzureSqlVirtualMachineCmdletBase.cs

```diff
@@ -0,0 +1,143 @@
+
+        /// <summary>
+        /// Gets an entity from the service
+        /// </summary>
+        /// <returns>A model object</returns>
+        protected abstract M GetEntity();
+
+        /// <summary>
+        /// Updates the given model element with the cmdlet specific operation 
+        /// </summary>
+        /// <param name="model">A model object</param>
+        protected virtual M ApplyUserInputToModel(M model) { return model; }
+
+        /// <summary>
+        /// This method is responsible to call the right API in the communication layer that will eventually send the information in the 
+        /// object to the REST endpoint
+        /// </summary>
+        /// <param name="entity">The model object with the data to be sent to the REST endpoints</param>
+        protected virtual M PersistChanges(M entity) { return default(M); }
+
+        /// <summary>
+        /// Returns true if the model object that was constructed by this cmdlet should be written out
+        /// </summary>
+        /// <returns>True if the model object should be written out, False otherwise</returns>
+        protected virtual bool WriteResult() { return true; }
+
+        /// <summary>
+        /// Creation and initialization of the ModelAdapter object
+        /// </summary>
+        /// <returns>An initialized and ready to use ModelAdapter object</returns>
+        protected abstract A InitModelAdapter();
+
+        /// <summary>
+        /// Transforms the given model object to be an object that is written out
+        /// </summary>
+        /// <param name="model">The about to be written model object</param>
+        /// <returns>The prepared object to be written out</returns>
+        protected virtual object TransformModelToOutputObject(M model)
+        {
+            return model;
+        }
+
+        /// <summary>
+        /// Get the name of the action to be displayed in the request of confirmation for this cmdlet
+        /// </summary>
+        /// <returns>The name of the cmdlet that is being executed</returns>
+        protected virtual string GetConfirmActionProcessMessage()
+        {
+            return MyInvocation.MyCommand.Name.Split('-')[0].ToLower();
+        }
+
+        /// <summary>
+        /// Parse the input provided to the cmdlet
+        /// </summary>
+        protected virtual void ParseInput() { }
+
+        /// <summary>
+        /// Executes the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            ModelAdapter = InitModelAdapter();
+            ParseInput();
+            M model = GetEntity();
+            M updatedModel = ApplyUserInputToModel(model);
+            M responseModel = default(M);
+            ConfirmAction(GetConfirmActionProcessMessage(), GetResourceId(updatedModel), () =>
+            {
+                responseModel = PersistChanges(updatedModel);
+            });
+
+            if (responseModel != null)
```

Comment:
NIT: Can you please add some comments to this if...else to say what is happening here